src/Features/CSharp/Portable/ConvertToRecord/ConvertToRecordCommon.cs

```diff
@@ -109,9 +97,9 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
             // remove properties we're bringing up to positional params
             // or keep them as overrides and link the positional param to the original property
-            foreach (var result in positionalParameterInfos)
+            foreach (var result in positionalParameterInfos.Where(prop => !prop.IsInherited))
             {
-                var property = result.Declaration;
+                var property = result.Declaration!;
```

Comment:
If you werre to switch the foreach to just say "if (prop.IsInherited) continue;" you can then annotate this to avoid the !.